### PR TITLE
Refactor #129: Refactoring the Event System

### PR DESCRIPTION
**Pull request by sinnaj-r**
Merge after  #138.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,8 +36,7 @@ Algernon allows for keyboard only interaction - no need to use your mouse!
 - Search global variables
 - Print and evaluate Smalltalk expressions 
 - Action verbs for filtering and quick access (e.g. `run SomeTest` will only search for tests and set a default action)
-- Full text search in classes
-- Save and/or quit the image
```
Comment (BenBals) [inline README.md:40]: Why exactly did you remove this? Or is this a merge thing?

In any case: Please change it.

File: packages/Algernon.package/ALGClassType.class/instance/matchesComment..st [hunk 2]
```diff
@@ -0,0 +1,3 @@
+accessing
+matchesComment: aString
```
Comment (BenBals) [inline packages/Algernon.package/ALGClassType.class/instance/matchesComment..st]: I'm not sure we need this method anymore. Could you check?

File: packages/Algernon.package/ALGItem.class/instance/matchesComment..st [hunk 3]
```diff
@@ -3,4 +3,5 @@ matchesComment: aString
 
 	self algerType hasComment
```
Comment (BenBals) [inline packages/Algernon.package/ALGItem.class/instance/matchesComment..st]: same here. I'm not sure we need this anymore.

**Review by BenBals — APPROVED** [review]
Looks good to me. The comments I have left are more related to #138, but since there is some weird branching going on here, I thought I might as well look at it.

**Comment by BenBals** [comment]
# Note that this PR may only be merged after #138.

File: README.md (hunk 1 above)
Comment (sinnaj-r) [reply README.md:40]: This is a Merge Error; I restored it.

File: packages/Algernon.package/ALGClassType.class/instance/matchesComment..st (hunk 2 above)
Comment (sinnaj-r) [reply packages/Algernon.package/ALGClassType.class/instance/matchesComment..st]: We don't; I removed it.

File: packages/Algernon.package/ALGItem.class/instance/matchesComment..st (hunk 3 above)
Comment (sinnaj-r) [reply packages/Algernon.package/ALGItem.class/instance/matchesComment..st]: We don't; I removed it.

**Comment by BenBals** [comment]
Awesome, thanks for your changes. We will merge this as soon as #138 is merged!

**Pull request merged**